### PR TITLE
Add picnic mechanics to Cap Pikachu

### PR DESCRIPTION
**Pull request by dot-Comfey**
https://www.smogon.com/forums/threads/cap-pikachu-should-be-able-to-learn-egg-moves.3742218/

Reading the conversation at #9888 indicates that the favorable solution is to just hardcode the egg moves into the learnsets of Pokemon like Cap Pikachu. I'm not sure if we still want that or if we eventually wanted to replace it with a generic implementation, but since Cap Pikachu's egg moves are still illegal, some fix is warranted.

Instead of trying to stick in just the egg moves, I'm allowing the entire learnset of the base evolution to be taken, but filtering out everything but the egg moves when going through the learnset of the base evolution. (Side note: checkingMoves is added as a parameter to learnsetParent because it's called by validatePokemonGo to find valid base forms rather than for learnset reasons)

### CODE REVIEW

File: test/sim/team-validator/breeding.js [hunk 1]
```diff
@@ -233,6 +233,7 @@ describe('Team Validator', function () {
 		team = [
 			{species: 'ursalunabloodmoon', ability: 'mindseye', moves: ['yawn', 'bellydrum'], evs: {hp: 1}},
 			{species: 'greninjabond', ability: 'battlebond', moves: ['counter', 'switcheroo'], evs: {hp: 1}},
+			{species: 'pikachualola', ability: 'static', moves: ['wish', 'fakeout'], evs: {hp: 1}},
```
Comment (urkerab) [inline test/sim/team-validator/breeding.js:236]: Does this not fix the other two cases as well? (If it does then shouldn't you remove the learnset changes from #9888?)

File: sim/team-validator.ts [hunk 2]
```diff
@@ -2421,6 +2422,14 @@ export class TeamValidator {
 				}
 			}
 
+			if (checkingPrevo && !originalSpecies.prevo) {
+				if (dex.gen >= 9) {
+					eggMovesOnly = true;
+				} else {
+					break;
+				}
```
Comment (urkerab) [inline sim/team-validator.ts:2430]:
```suggestion
				if (dex.gen < 9) {
					break;
				}
				eggMovesOnly = true;
```

File: sim/team-validator.ts [hunk 3]
```diff
@@ -2468,6 +2477,8 @@ export class TeamValidator {
 					continue;
 				}
 
+				if (eggMovesOnly && learned.charAt(1) !== 'E') continue;
```
Comment (urkerab) [inline sim/team-validator.ts:2480]: Another approach would be to put the `checkingPrevo && !originalSpecies.prevo` check here and remove the `eggMovesOnly` variable.

File: sim/dex-species.ts [hunk 4]
```diff
@@ -623,7 +623,7 @@ export class DexSpecies {
 		return out;
 	}
 
-	learnsetParent(species: Species) {
+	learnsetParent(species: Species, checkingMoves?: boolean) {
```
Comment (KrisXV) [inline sim/dex-species.ts:626]:
```suggestion
	learnsetParent(species: Species, checkingMoves = false) {
```

File: test/sim/team-validator/breeding.js (hunk 1 above)
Comment (dot-Comfey) [reply test/sim/team-validator/breeding.js:236]: It does, but I wanted to make sure it's fine to go ahead and remove the learnset changes. It seems like there's no objection here, so I will go ahead and do that.

**Comment by DaWoblefet** [comment]
Thanks dot-Comfey!

**Comment by urkerab** [comment]
`getMovePool` needs a similar change; it's currently adding all of Teddiursa's moves to Ursaluna-Bloodmoon's move pool instead of just its egg moves.

**Pull request merged**